src/compile/references.js

```diff
@@ -270,9 +270,11 @@ module.exports = function(Velocity, utils) {
 
           var that = this;
 
-          baseRef.eval = function() {
-            return that.eval.apply(that, arguments);
-          };
+          if(typeof baseRef === 'object'){
```

Comment:
Good catch!

 If you want to be efficient, you might check for *falsy* values before calling typeof. That way, if `baseRef` is *falsy* a typeof evaluation is not necessary, as objects are always *truthy*.
```
if (baseRef && typeof baseRef === 'object') {
```